### PR TITLE
Vendor AMQP at commit 52942afb1a4b48a1d8af7a3ac37d3211661e3423

### PR DESCRIPTION
**Pull request by LarryOsterman**
Vendored copy of azure-uamqp-c to port azure-uamqp-python fixes and make other changes.

All of the changes under the "vendor" directory are simply a clone of the uAMQP repo and can be ignored in this PR as being essentially out-of-scope.

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/core/azure-core-amqp/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_types.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#define ANQP_TYPES_H
+
+#include <stddef.h>
+#include "azure_macro_utils/macro_utils.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#define AMQP_TYPE_VALUES \
```
Comment (antkmsft) [inline sdk/core/azure-core-amqp/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_types.h:14]: How visible would this be to the SDK users? Are we going to leak any of the stuff that's defined in these headers (not this specific macro, but anything)?

If our headers do only forward-declare AMQP types at best, and it is only the SDK cpp files code that see this, then everything is easy and I am very happy.

Comment (antkmsft) [reply sdk/core/azure-core-amqp/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_types.h:14]: I.e. if you would produce an ApiView from this, without additional tweaks to ignore certain stuff, what would be visible?

Comment (LarryOsterman) [reply sdk/core/azure-core-amqp/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_types.h:14]: uAMQP is a vendored version of azure-uamqp-c.

It is NOT an azure SDK. 

It is literally a replacement for the azure-uamqp-c vcpkg package.

There is a fair amount of work in the azure-core-amqp to ensure that there is no leakage.

**Comment by LarryOsterman** [comment]
Ping.

This one's relatively small - the entire "vendor" directory can be ignored since it is just a copy of uAMQP as of #52942afb1a4b48a1d8af7a3ac37d3211661e3423

**Review by antkmsft — APPROVED** [review]

**Pull request merged**